### ISSUE TITLE
Run clippy and fix things

### ISSUE THREAD

**Issue by mre** [mre/mos6502#6]
I haven't tried it, but I assume if we ran clippy on this repo, we could find a few things to fix. Might be a nice exercise for Rust beginners.
Up for grabs, if anyone wants to give it a shot.

**Comment by kzzt** [comment]
Hi! I ran clippy on the repo and the first thing that stands out is 

`[[bin]]
# This will look in src/bin/mos6502.rs
name = "mos6502.rs"`

main.rs is the only file in src/bin

so I changed Cargo.toml to have `[[bin]]
# This will look in src/bin/mos6502.rs
name = "main"`

Also I added `autobins = false` to [package] in response to the following warning: 

> warning: An explicit [[bin]] section is specified in Cargo.toml which currently
disables Cargo from automatically inferring other binary targets.
This inference behavior will change in the Rust 2018 edition and the following
files will be included as a binary target:

>* X:\Users\xyz\git_projects\mos6502\src\main.rs

>This is likely to break cargo build or cargo test as these files may not be
ready to be compiled as a binary target today. You can future-proof yourself
and disable this warning by adding `autobins = false` to your [package]
section. You may also move the files to a location where Cargo would not
automatically infer them to be a target, such as in subfolders.

Apologies if this is nonsense- I'm very new to both rust and github, but saw this issue as an opportunity to get my feet wet without screwing up. Thanks.

**Comment by mre** [comment]
Heyo @kzzt,

welcome to Rust! The community where everything goes. 🌈 Glad to have you on the team. 🦀 
Nice that you're tackling this.

> main.rs is the only file in src/bin

That's true. My fault. As far as I understand it, renaming that file to `src/bin/mos6502.rs` without any other changes should make the warning go away. 

> Also I added autobins = false to [package] in response to the following warning: [...]

That was the logical next step from what you did. Rust tells us that it dected `main.rs` in the bin folder and that without a `[[bin]]` section in our `Cargo.toml`, it would just assume that's a binary we'd like to compile. In the `[[bin]]` section we mistakenly specified a (nonexistent) binary however. And it was unsure, whether the `main.rs` would be something that is there by accident or on purpose, since it was never mentioned in the `[[bin]]` section.

I looked at `main.rs` and it doesn't make a lot of sense to me. Ported that over with the fork. It looks more like an integration test to me. So I added it to the `README.md` as a usage-example, which also gets executed with [skeptic](https://github.com/budziq/rust-skeptic/) when running `cargo test`.

> Apologies if this is nonsense- I'm very new to both rust and github, but saw this issue as an opportunity to get my feet wet without screwing up. Thanks.

That's great! Keep the pull requests coming. I'm sure there are many more odd things in this library. Just open more issues and we can all discuss them here. 👍
